### PR TITLE
chore: rearrange code before refactoring

### PR DESCRIPTION
**Pull request by ttypic**
It's hard to analyze the code when it is not arranged properly based on Java guidelines 

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling for proxy configuration in HTTP connections.
	- Improved clarity in WebSocket connection handling and close operations.

- **Bug Fixes**
	- Refined error handling for server responses, including authentication headers.

- **Refactor**
	- Code organization and readability improvements in both HTTP and WebSocket handling.
	- Method renaming and restructuring for better clarity and functionality.
	- Updated Gradle and wrapper scripts for improved build process efficiency.

### CODE REVIEW

File: lib/src/main/java/io/ably/lib/transport/WebSocketTransport.java [hunk 1]
```diff
@@ -65,24 +76,24 @@ public void connect(ConnectListener connectListener) {
...
-            synchronized(this) {
+            synchronized (this) {
                 wsConnection = new WsClient(URI.create(wsUri), this::receive);
-                if(isTls) {
+                if (isTls) {
                     SSLContext sslContext = SSLContext.getInstance("TLS");
-                    sslContext.init( null, null, null );
+                    sslContext.init(null, null, null);
                     SafeSSLSocketFactory factory = new SafeSSLSocketFactory(sslContext.getSocketFactory());
                     wsConnection.setSocketFactory(factory);
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply lib/src/main/java/io/ably/lib/transport/WebSocketTransport.java:89]: I also support this 👍

File: lib/src/main/java/io/ably/lib/http/HttpCore.java [hunk 2]
```diff
@@ -77,21 +114,21 @@ public HttpCore(ClientOptions options, Auth auth, PlatformAgentProvider platform
...
         boolean renewPending = true, proxyAuthPending = true;
-        if(requireAblyAuth) {
+        if (requireAblyAuth) {
             authorize(false);
         }
-        while(true) {
+        while (true) {
             try {
                 return httpExecute(url, getProxy(url), method, headers, requestBody, true, responseHandler);
-            } catch(AuthRequiredException are) {
```
Comment (sacOO7) [inline lib/src/main/java/io/ably/lib/http/HttpCore.java:86]: Can we rename `are` to `authReqEx`

File: lib/src/main/java/io/ably/lib/http/HttpCore.java [hunk 3]
```diff
@@ -152,20 +189,21 @@ void authorize(boolean renew) throws AblyException {
     public <T> T httpExecute(URL url, Proxy proxy, String method, Param[] headers, RequestBody requestBody, boolean withCredentials, ResponseHandler<T> responseHandler) throws AblyException {
         HttpURLConnection conn = null;
         try {
-            conn = (HttpURLConnection)url.openConnection(proxy);
+            conn = (HttpURLConnection) url.openConnection(proxy);
             boolean withProxyCredentials = (proxy != Proxy.NO_PROXY) && (proxyAuth != null);
             return httpExecute(conn, method, headers, requestBody, withCredentials, withProxyCredentials, responseHandler);
-        } catch(IOException ioe) {
+        } catch (IOException ioe) {
```
Comment (sacOO7) [inline lib/src/main/java/io/ably/lib/http/HttpCore.java:195]: rename `ioe` to `ioExecption`

File: lib/src/main/java/io/ably/lib/http/HttpCore.java [hunk 4]
```diff
@@ -235,9 +278,9 @@ <T> T httpExecute(HttpURLConnection conn, String method, Param[] headers, Reques
                         Log.v(TAG, "  " + entry.getKey() + ": " + val);
             }
 
-            if(options instanceof DebugOptions) {
-                rawHttpListener = ((DebugOptions)options).httpListener;
-                if(rawHttpListener != null) {
+            if (options instanceof DebugOptions) {
+                rawHttpListener = ((DebugOptions) options).httpListener;
+                if (rawHttpListener != null) {
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply lib/src/main/java/io/ably/lib/http/HttpCore.java:283]: Yeah, not sure why we are not providing other way to create `UUID`

File: lib/src/main/java/io/ably/lib/http/HttpCore.java [hunk 5]
```diff
@@ -420,7 +469,8 @@
         InputStream is = null;
         try {
             is = connection.getInputStream();
-        } catch (Throwable e) {}
+        } catch (Throwable e) {
+        }
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply lib/src/main/java/io/ably/lib/http/HttpCore.java:473]: Not sure why exception message is not logged using logger

File: lib/src/main/java/io/ably/lib/http/HttpCore.java [hunk 6]
```diff
@@ -433,7 +483,8 @@ private Response readResponse(HttpURLConnection connection) throws IOException {
             if (is != null) {
                 try {
                     is.close();
-                } catch (IOException e) {}
+                } catch (IOException e) {
```
Comment (sacOO7) [inline lib/src/main/java/io/ably/lib/http/HttpCore.java:486]: Exception error should be logged

File: lib/src/main/java/io/ably/lib/http/HttpCore.java [hunk 7]
```diff
@@ -77,21 +114,21 @@ public HttpCore(ClientOptions options, Auth auth, PlatformAgentProvider platform
...
             authorize(false);
         }
-        while(true) {
+        while (true) {
             try {
                 return httpExecute(url, getProxy(url), method, headers, requestBody, true, responseHandler);
-            } catch(AuthRequiredException are) {
-                if(are.authChallenge != null && requireAblyAuth) {
-                    if(are.expired && renewPending) {
+            } catch (AuthRequiredException are) {
```
Comment (sacOO7) [inline lib/src/main/java/io/ably/lib/http/HttpCore.java:123]: Exception message should be logged. This is true for all other exceptions in other places. Helpful for debugging

File: lib/src/main/java/io/ably/lib/http/HttpCore.java [hunk 8]
```diff
@@ -302,68 +346,72 @@ private <T> T handleResponse(HttpURLConnection conn, boolean credentialsIncluded
...
                 try {
                     ErrorResponse errorResponse = ErrorResponse.fromJSON(bodyText);
-                    if(errorResponse != null) {
+                    if (errorResponse != null) {
                         error = errorResponse.error;
                     }
-                } catch(JsonParseException jse) {
+                } catch (JsonParseException jse) {
                     /* error pages aren't necessarily going to satisfy our Accept criteria ... */
                     System.err.println("Error message in unexpected format: " + bodyText);
```
Comment (sacOO7) [inline lib/src/main/java/io/ably/lib/http/HttpCore.java:354]: Not sure why standard logger is not being used to log bodytext and exception.

Comment (sacOO7) [reply lib/src/main/java/io/ably/lib/http/HttpCore.java:354]: This should be identified and replaced with standard loggers wherever possible

File: lib/src/main/java/io/ably/lib/http/HttpCore.java [hunk 9]
```diff
@@ -77,21 +114,21 @@ public HttpCore(ClientOptions options, Auth auth, PlatformAgentProvider platform
...
+            } catch (AuthRequiredException are) {
+                if (are.authChallenge != null && requireAblyAuth) {
+                    if (are.expired && renewPending) {
                         authorize(true);
                         renewPending = false;
                         continue;
                     }
                 }
-                if(are.proxyAuthChallenge != null && proxyAuthPending && proxyAuth != null) {
+                if (are.proxyAuthChallenge != null && proxyAuthPending && proxyAuth != null) {
```
Comment (sacOO7) [inline lib/src/main/java/io/ably/lib/http/HttpCore.java:131]: `proxyAuthChallenge` is confusing. Can we replace it with some other keyword?

File: lib/src/main/java/io/ably/lib/http/HttpCore.java [hunk 10]
```diff
@@ -191,32 +229,37 @@ <T> T httpExecute(HttpURLConnection conn, String method, Param[] headers, Reques
             if (authHeader == null && auth != null) {
                 authHeader = auth.getAuthorizationHeader();
             }
-            if(withCredentials && authHeader != null) {
+            if (withCredentials && authHeader != null) {
                 conn.setRequestProperty(HttpConstants.Headers.AUTHORIZATION, authHeader);
                 credentialsIncluded = true;
             }
-            if(withProxyCredentials && proxyAuth.hasChallenge()) {
+            if (withProxyCredentials && proxyAuth.hasChallenge()) {
```
Comment (sacOO7) [inline lib/src/main/java/io/ably/lib/http/HttpCore.java:236]: `hasChallenge` term is confusing, also no documentation for the same

**Comment by ttypic** [comment]
@sacOO7 It's not refactoring :) it's just rearranging the code. The code order was wrong, I made it right, based on official Java specification. We have static initializers, then static fields, methods, public fields, private fields, constructor, then public methods, private methods. Everything else is out of the scope of this PR. I haven't change a single line of code

**Review by sacOO7 — APPROVED** [review]
Lgtm

**Pull request merged**